File: Terminus-Edition-2/aws-lambda-event-source-mapping-recovery/environment/src/mapping.py
from __future__ import annotations
from typing import Any

def active_mapping(mapping: dict[str, Any], queues: dict[str, Any]) -> dict[str, Any]:
    enabled = bool(mapping.get("enabled"))
    expected_arn = queues.get("expected_active_queue_arn")
    source_arn = mapping.get("event_source_arn")
    active = enabled and source_arn == expected_arn
    return {"uuid": mapping.get("uuid"), "enabled": enabled, "active": active, "function_name": mapping.get("function_name"), "event_source_arn": source_arn, "expected_event_source_arn": expected_arn, "old_queue_arn": queues.get("old_queue_arn"), "batch_size": mapping.get("batch_size"), "function_response_types": mapping.get("function_response_types", [])}

File: Terminus-Edition-2/aws-lambda-fix84-pack/environment/simulator/mapping.go
package main

func activeMapping(mapping, queues map[string]any) map[string]any {
	enabled, _ := mapping["enabled"].(bool)
	expectedARN := stringValue(queues["expected_active_queue_arn"])
	sourceARN := stringValue(mapping["event_source_arn"])
	responseTypes := mapping["function_response_types"]
	if responseTypes == nil {
		responseTypes = []any{}
	}
	return map[string]any{
		"uuid":                      mapping["uuid"],
		"enabled":                   enabled,
		"active":                    enabled && sourceARN == expectedARN,
		"function_name":             mapping["function_name"],
		"event_source_arn":          mapping["event_source_arn"],
		"expected_event_source_arn": queues["expected_active_queue_arn"],
		"old_queue_arn":             queues["old_queue_arn"],
		"batch_size":                mapping["batch_size"],
		"function_response_types":   responseTypes,
	}
}
